pydantic/_internal/_typing_extra.py

```diff
@@ -445,3 +448,11 @@ def evaluate_fwd_ref(
         ref: ForwardRef, globalns: dict[str, Any] | None = None, localns: dict[str, Any] | None = None
     ) -> Any:
         return ref._evaluate(globalns=globalns, localns=localns, recursive_guard=frozenset())
+
+
+def is_dataclass(_cls: type[Any]) -> TypeGuard[type[StandardDataclass]]:
+    # The dataclasses.is_dataclass function doesn't seem to provide TypeGuard functionality,
+    # so I created this convenience function
+    import dataclasses
+
+    return dataclasses.is_dataclass(_cls)
```

Comment:
How is this different from `is_builtin_dataclass` above?

Reply:
it returns True for pydantic dataclasses as well; is_builtin_dataclass is only true for _non_ pydantic dataclasses.

Reply:
Gotcha. Maybe it should be:

```suggestion
def is_dataclass(_cls: type[Any]) -> TypeGuard[type[StandardDataclass] | type[PydanticDataclass]]:
    # The dataclasses.is_dataclass function doesn't seem to provide TypeGuard functionality,
    # so I created this convenience function
    import dataclasses

    return dataclasses.is_dataclass(_cls)
```

?

Reply:
a `PydanticDataclass` is already a `StandardDataclass` so I don't think the union is semantically meaningful, but if you think it conveys intent to humans better we could do that